src/app/groups/group-set-manager/group-set-manager.component.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'group-set-manager',
+  templateUrl: './group-set-manager.component.html',
+  styleUrls: ['./group-set-manager.component.scss'],
+})
+export class GroupSetManagerComponent {
+  @Input() unit: any;
+  @Input() unitRole: any;
+  @Input() project: any;
+  @Input() selectedGroup: any;
+  @Input() selectedStudent: any;
+  @Input() selectedGroupSet: any;
+  @Input() showGroupSetSelector: boolean;
+
+  constructor() {}
+
+  newGroupSelected(event: any) {
+    if (this.selectedGroupSet) {
+      this.selectedGroup = this.selectedGroupSet.groups[0];
+    }
+    console.log('New group selected:', this.selectedGroup);
```

Comment:
No need for console logs. If the original coffee file showed alerts or success messages upon completing actions, follow that